libsql-sqlite3/src/wal.c

```diff
@@ -1729,6 +1766,51 @@ static int walIteratorInit(Wal *pWal, u32 nBackfill, WalIterator **pp){
   return rc;
 }
 
+static int walIteratorRevInit(Wal *pWal, u32 nBackfill, struct WalIteratorRev *p, u32 mxSafeFrame, int getMissingFrame){
+    WalIterator *pIter;
+    u32 *frames;
+    u32 iFrame, iPageno;
+    int rc;
+
+    frames = (u32*)sqlite3MallocZero((pWal->hdr.mxFrame + 1) * sizeof(u32));
+    if (!frames) return SQLITE_NOMEM_BKPT;
+    rc = walIteratorInit(pWal, nBackfill, &pIter);
+    if (rc || !pIter) {
+        sqlite3_free(frames);
+        return rc;
+    }
+
+    while (walIteratorNext(pIter, &iPageno, &iFrame) == 0) {
+        /*
+         * If we get a page with a frame_no greater than mxSafeFrame, and getMissingFrame is true,
```

Comment:
ok, one more nitpick on a comment and variable name. I remember we talked about this design, but `getMissingFrame` is misleading, this is not really a "missing" frame, we want to eagerly look for an older frame if the newest one is "too new". `allow_older_frame_versions`, `eager_frame_lookup`? I'm trying to come up a name which makes it clear that the default behavior is that we either get the newest frame or nothing, while in this case we explicitly ask for a frame number that used to be newest in `mxFrame` point in time.